File: update_address.py
import os
import glob

html_files = glob.glob("**/*.html", recursive=True)

old_addr1 = "9803 Highway 242, Suite 200-203<br>Conroe, TX 77385"
new_addr1 = "27128 Hanna Road<br>Conroe, TX 77385"

old_addr2 = "9803 Highway 242, Suite 200-203 Conroe TX 77385"
new_addr2 = "27128 Hanna Road, Conroe, TX 77385"

# Also check city page footers with old placeholder 1301 Fannin St
old_fannin = "1301 Fannin St"

updated_files = []

for file_path in html_files:
    with open(file_path, "r", encoding="utf-8", errors="ignore") as f:
        content = f.read()
    
    new_content = content
    if old_addr1 in new_content:
        new_content = new_content.replace(old_addr1, new_addr1)
    if old_addr2 in new_content:
        new_content = new_content.replace(old_addr2, new_addr2)
    
    # Update city page address footers if present
    lines = new_content.splitlines()
    modified_lines = []
    for line in lines:
        if "1301 Fannin St" in line:
            # Replace placeholder city address with official headquarters address
            line = line.split("</i>")[0] + "</i> 27128 Hanna Road, Conroe, TX 77385</li>"
        modified_lines.append(line)
    
    new_content = "\n".join(modified_lines)
    
    if new_content != content:
        with open(file_path, "w", encoding="utf-8") as f:
            f.write(new_content)
        updated_files.append(file_path)

print(f"Updated {len(updated_files)} HTML files:")
for u in updated_files:
    print(f" - {u}")

File: update_address.js
const fs = require('fs');
const path = require('path');

function getFiles(dir, fileList = []) {
  const files = fs.readdirSync(dir);
  files.forEach(file => {
    const filePath = path.join(dir, file);
    if (fs.statSync(filePath).isDirectory()) {
      getFiles(filePath, fileList);
    } else if (file.endsWith('.html')) {
      fileList.push(filePath);
    }
  });
  return fileList;
}

const htmlFiles = getFiles(__dirname + '/..');

const oldAddr1 = "9803 Highway 242, Suite 200-203<br>Conroe, TX 77385";
const newAddr1 = "27128 Hanna Road<br>Conroe, TX 77385";

const oldAddr2 = "9803 Highway 242, Suite 200-203 Conroe TX 77385";
const newAddr2 = "27128 Hanna Road, Conroe, TX 77385";

let updatedCount = 0;

htmlFiles.forEach(file => {
  let content = fs.readFileSync(file, 'utf8');
  let updated = false;

  if (content.includes(oldAddr1)) {
    content = content.replaceAll(oldAddr1, newAddr1);
    updated = true;
  }
  if (content.includes(oldAddr2)) {
    content = content.replaceAll(oldAddr2, newAddr2);
    updated = true;
  }

  // Replace placeholder city addresses if present
  if (content.includes('1301 Fannin St')) {
    content = content.replace(/1301 Fannin St, [^<]+<\/li>/g, '27128 Hanna Road, Conroe, TX 77385</li>');
    updated = true;
  }

  if (updated) {
    fs.writeFileSync(file, content, 'utf8');
    console.log(`Updated: ${path.relative(__dirname + '/..', file)}`);
    updatedCount++;
  }
});

console.log(`Total HTML files updated with new address: ${updatedCount}`);
